### ISSUE TITLE
CMake 2.8.12 thinks Intel 15.0.2 build is actually 15.0.0.20150121 and is not supported by Kokkos

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#2160]
**CC:** @trilinos/framework, @trilinos/kokkos

## Description

I was looking over the Intel builds for Trilinos as part of #2142 and I noticed that the build [Linux-intel-15.0.2-MPI_RELEASE_DEV_DownStream_ETI_SERIAL-OFF_OPENMP-ON_PTHREAD-OFF_CUDA-OFF_COMPLEX-OFF build](https://testing.sandia.gov/cdash/index.php?project=Trilinos&parentid=3329931) that claims to be for Intel 15.0.2 is actually Intel 15.0.0.20150121 and that version is **not** supported by Kokkos as shown in the [configure output](https://testing.sandia.gov/cdash/viewConfigure.php?buildid=3329935):

```
- CMAKE_VERSION='2.8.12.2'
[...]
-- MPI_EXEC='/projects/sems/install/rhel6-x86_64/sems/compiler/intel/15.0.2/openmpi/1.8.7/bin/mpiexec'
-- MPI_EXEC='/projects/sems/install/rhel6-x86_64/sems/compiler/intel/15.0.2/openmpi/1.8.7/bin/mpiexec'
-- CMAKE_C_COMPILER_ID='Intel'
-- CMAKE_C_COMPILER_VERSION='15.0.0.20150121'
-- CMAKE_CXX_COMPILER_ID='Intel'
-- CMAKE_CXX_COMPILER_VERSION='15.0.0.20150121'
CMake Error at packages/kokkos/cmake/kokkos_functions.cmake:72 (message):
  Compiler not supported by Kokkos.  Required compiler versions:

      Clang      3.5.2 or higher
      GCC        4.8.4 or higher
      Intel     15.0.2 or higher
      NVCC      7.0.28 or higher
      PGI         17.1 or higher

Call Stack (most recent call first):
  cmake/ProjectCompilerPostConfig.cmake:13 (set_kokkos_cxx_compiler)
  /home/jenkins/slave/workspace/Kokkos_Trilinos_packages/Trilinos/cmake/tribits/core/package_arch/TribitsGlobalMacros.cmake:1820 (INCLUDE)
  /home/jenkins/slave/workspace/Kokkos_Trilinos_packages/Trilinos/cmake/tribits/core/package_arch/TribitsProjectImpl.cmake:188 (TRIBITS_SETUP_ENV)
  /home/jenkins/slave/workspace/Kokkos_Trilinos_packages/Trilinos/cmake/tribits/core/package_arch/TribitsProject.cmake:93 (TRIBITS_PROJECT_IMPL)
  CMakeLists.txt:93 (TRIBITS_PROJECT)

-- Configuring incomplete, errors occurred!
```

This configure failure for Kokkos disables Kokkos in all downstream packages and as a result of Kokkos being disabled (and Teuchos being disabled as well due to #2128), the only packages that build any libraries and tests and run any tests are Sacado, Epetra, and Zoltan.

And the problem seems to be that SEMS thinks this is actually Intel 15.0.2 given the SEMS TPL paths like shown in the configure output above that shows the path `/projects/sems/install/rhel6-x86_64/sems/compiler/intel/15.0.2/[...]`.

## Related Issues

* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to: #2142
* Part of 
* Composed of

**Comment by bartlettroscoe** [comment]
This is strange, but the Intel compiler loaded with `module load sems-intel/15.0.2` says that it is version `15.0.2 20150121` as shown by:

```
$ module load sems-env
$ module load sems-intel/15.0.2 

$ which $CC
/projects/sems/install/rhel6-x86_64/sems/compiler/intel/15.0.2/base/bin/icc

$ $CC --version
icc (ICC) 15.0.2 20150121
Copyright (C) 1985-2015 Intel Corporation.  All rights reserved.

$ which $CXX
/projects/sems/install/rhel6-x86_64/sems/compiler/intel/15.0.2/base/bin/icpc

$ $CXX --version
icpc (ICC) 15.0.2 20150121
Copyright (C) 1985-2015 Intel Corporation.  All rights reserved.
```

Therefore, this makes me think that CMake 3.5.2 is reading the version wrong for this compiler.  I wonder if this is actually a CMake defect?

**Comment by bartlettroscoe** [comment]
So it seems seems that CMake 3.5.2 is able to parse the version string properly as shown by:

```
$ cd $HOME/Trilinos.base/BUILDS/INTEL-15.0.2/SERIAL_DEBUG/

$ module purge
$ module load sems-env
$ module load sems-intel/15.0.2 
$ module load sems-cmake/3.5.2 

$ cmake -DTrilinos_ENABLE_KokkosCore=ON -DTrilios_ENABLE_TESTS=ON ../../../Trilinos \
  &> configure.out
```

and the configure output shows:

```
-- CMAKE_VERSION='3.5.2'
[...]
-- The C compiler identification is Intel 15.0.2.20150121
-- Check for working C compiler: /projects/sems/install/rhel6-x86_64/sems/compiler/intel/15.0.2/base/bin/icc
-- Check for working C compiler: /projects/sems/install/rhel6-x86_64/sems/compiler/intel/15.0.2/base/bin/icc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- CMAKE_C_COMPILER_ID='Intel'
-- CMAKE_C_COMPILER_VERSION='15.0.2.20150121'
-- The CXX compiler identification is Intel 15.0.2.20150121
-- Check for working CXX compiler: /projects/sems/install/rhel6-x86_64/sems/compiler/intel/15.0.2/base/bin/icpc
-- Check for working CXX compiler: /projects/sems/install/rhel6-x86_64/sems/compiler/intel/15.0.2/base/bin/icpc -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- CMAKE_CXX_COMPILER_ID='Intel'
-- CMAKE_CXX_COMPILER_VERSION='15.0.2.20150121'
-- The Fortran compiler identification is Intel 15.0.2.20150121
-- Check for working Fortran compiler: /projects/sems/install/rhel6-x86_64/sems/compiler/intel/15.0.2/base/bin/ifort
-- Check for working Fortran compiler: /projects/sems/install/rhel6-x86_64/sems/compiler/intel/15.0.2/base/bin/ifort  -- works
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Checking whether /projects/sems/install/rhel6-x86_64/sems/compiler/intel/15.0.2/base/bin/ifort supports Fortran 90
-- Checking whether /projects/sems/install/rhel6-x86_64/sems/compiler/intel/15.0.2/base/bin/ifort supports Fortran 90 -- yes
-- Using the legacy tribits build because KOKKOS_ARCH not set
-- 
-- ****************** Kokkos Settings ******************
-- Execution Spaces
--   Device Parallel: None
--     Host Parallel: None
--       Host Serial: Serial
-- 
-- Architectures:
--     None
-- 
-- Enabled options
--   KOKKOS_ENABLE_PROFILING
-- 
-- Final kokkos settings variable:
--   env;KOKKOS_SRC_PATH=/ascldap/users/rabartl/Trilinos.base/Trilinos/packages/kokkos;KOKKOS_PATH=/ascldap/users/rabartl/Trilinos.base/Trilinos/packages/kokkos;KOKKOS_INSTALL_PATH=/usr/local;KOKKOS_ARCH=None;KOKKOS_DEVICES=Serial;KOKKOS_DEBUG=no;KOKKOS_OPTIONS=disable_dualview_modify_check
-- *****************************************************
```

Therefore, this problem can be fixed by simply upgrading CMake used for this build.

**Comment by bartlettroscoe** [comment]
@trilinos/framwork team,

If you accept PR #2162 and set `export CMAKE_SUFFIX=cmake/3.5.2` in the SRN Jenkins job `Kokkos_Trilinos_packages`, then it should fix the Kokkos configuration problem described here.

**Comment by crtrott** [comment]
Hm maybe we need to modify the Kokkos check anyway to keep supporting older cmake? That said we are up to 3.3 with our raw cmake support ...

**Comment by bartlettroscoe** [comment]
> Hm maybe we need to modify the Kokkos check anyway to keep supporting older cmake?

Perhaps but I doubt that any real customers are using something as old as CMake 2.8.12.  It seems like a lot of customers upgrade CMake pretty frequently.  And the real issue is that we are not testing Kokkos with Intel 15.0.0, we are testing it with 15.0.2.  I don't know if that is a big deal but or not but the current check is accurate supported by automated testing.  Saying that Kokkos supports Intel 15.0.0 without actually testing that seems like a bad idea.

> That said we are up to 3.3 with our raw cmake support ...

I think we should upgrade CMake as high as we can but the debate goes on in #1761.  There are just so many motivators for upgrading our minimum CMake version (and therefore upgrade the version of CMake used in automated testing).

**Comment by bartlettroscoe** [comment]
Intel 15 is old news.  Closing as "wontfix"
